pkg/daemon/cleanup/cleanup.go

```diff
@@ -92,18 +94,18 @@ func Deconfigure(cfg *rest.Config, ctx context.Context, configMapNamespace strin
 }
 
 // getMergedCMData attempts to get the latest and the version CM data specified by the node's version annotation
-// It returns the merged CM Data containing services and the watched environment variables
+// It returns the merged CM Data containing services, the watched environment variables, and if best effort is desired
 func getMergedCMData(ctx context.Context, cli client.Client,
-	configMapNamespace string, node *core.Node) (*servicescm.Data, error) {
+	configMapNamespace string, node *core.Node) (*servicescm.Data, bool, error) {
 	// get data from the latest services ConfigMap
 	latestCM, err := servicescm.GetLatest(cli, ctx, configMapNamespace)
 	if err != nil {
-		return nil, fmt.Errorf("cannot get latest services ConfigMap from namespace %s: %w",
+		return nil, false, fmt.Errorf("cannot get latest services ConfigMap from namespace %s: %w",
 			configMapNamespace, err)
 	}
 	latestCMData, err := servicescm.Parse(latestCM.Data)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	// attempt to get the ConfigMap specified by the version annotation
```

Comment:
Depending on what is done to this function to address the overload, breaking this inline func out into its own function could be a good idea.